### PR TITLE
Remove xml_doc

### PR DESCRIPTION
**Pull request by cannero**
xml_doc was removed in https://github.com/ionide/tree-sitter-fsharp/pull/78.

### CODE REVIEW

**Comment by cannero** [comment]
Hi, thanks for making the tree sitter mode for F#! I just tried it out and had an error because the `xml_doc` type was removed.

**Comment by cannero** [comment]
Sorry, maybe I'm also misunderstand how this repository is synchronized with the tree-sitter fsharp library. Is there a certain commit which should be used?

I just tried the newest tree-sitter version and there it seems that `type` was replaced with `simple_type`.

**Comment by KaranAhlawat** [comment]
Hi @cannero! Thanks for your interest in the package :)

They did indeed remove `xml_doc`, and I had also added a fix around the time they removed the node, I just forgot to push it upstream 😅 

That said, this package - and from what I understand, the F# tree-sitter grammar - are both very much still evolving, so breaking changes are quite common currently. 

The simple answer is that this repository isn't synchronized with the tree-sitter-fsharp library currently. I just visit the repo once in a while (or when I find a issue in it), look at the commit log, and fix anything that might need fixing.
This is in one part due to the fact that tree-sitter-fsharp has no marked releases or tags which you could use in Emacs, and sticking to a specific commit feels icky to me.

TLDR: Both tree-sitter-fsharp and this package are evolving and breakages are common. There are no releases yet that would provide a safety net of sorts. This will probably go on for a bit. Maybe we can ask the tree-sitter-fsharp maintainer to cut a release so we can stick to a grammar and not have things break out from under us 🤔

**Comment by KaranAhlawat** [comment]
Btw, for the PR, I'll just push the changes I had made locally, as it included a few enhancements as well.
And I'll look into the `type` breakage as soon as I have some time on my hands. In the meantime, any PRs are welcome! (indentation is sorely needed, it's non-existent).
